aiida/schedulers/datastructures.py

```diff
@@ -352,6 +352,20 @@ class JobTemplate(DefaultFieldsAttributeDict):  # pylint: disable=too-many-insta
     )
 
 
+class TemplateCodeInfo(DefaultFieldsAttributeDict):
+    """
+    Extra layer of code_info which parse code info from code and plugin setting and then
+    pass to job template
+    """
+    _default_fields = (
+        'cmdline_params',  # as a list of strings
+        'stdin_name',
+        'stdout_name',
+        'stderr_name',
+        'join_files',
+    )
+
+
```

Comment:
Heya, lets move away from using these bespoke data class solutions, and use python's built in solution: https://docs.python.org/3/library/dataclasses.html

```suggestion
from __future__ import annotations
from dataclasses import dataclass, field

@dataclass
class TemplateCodeInfo):
    """
    Extra layer of code_info which parse code info from code and plugin setting and then
    pass to job template
    """
    cmdline_params: list[str] = field(default_factory=list)
    stdin_name: None | str = None
    stdout_name: None | str = None
    stderr_name: None | str = None
    join_files: list[str] = field(default_factory=list)


```

Reply:
Cool, I like this style. Is this a newly introduced feature?

Reply:
Yep, from python 3.7+, and we are python>=3.8

Reply:
done. Thanks @chrisjsewell